### DISCUSSION TITLE
Use Vitest with Nextjs Config Environment

### DISCUSSION THREAD

**Discussion by nhattran998** [vitest-dev/vitest#1361] [Q&A] [answered]
Hi everyone, currently I am working on some NextJs projects. But the problem here is the way mocking data from `next/config`. It's the public and server runtime config environment.

Here is the detail error:
![image](https://user-images.githubusercontent.com/24296018/169865923-c0e23dbc-304e-46cd-b0f0-566039e728b3.png)

Init helper repo: https://github.com/harrytran998/vitest-next-env-runtime-conf
Online repo: https://stackblitz.com/edit/nextjs-wgtfhe?file=test/sum.test.ts

Here is my next config:

```
const nextConfig = {
  reactStrictMode: true,
  publicRuntimeConfig: {
    NODE_ENV: process.env.NEXT_PUBLIC_NODE_ENV,
    ENV: process.env.NEXT_PUBLIC_ENV,
  },
}

module.exports = nextConfig
```
Here it's my utility config

// ~utils/environment
```
import getConfig from 'next/config';
const { publicRuntimeConfig } = getConfig();

export const AppEnv = {
  NODE_ENV: publicRuntimeConfig.NODE_ENV,
  ENV: publicRuntimeConfig.ENV,
};
```

Now I create a basic test with vitest to calculate sum like this:

```
import { beforeAll, expect, test, vi } from 'vitest'
import { sum } from '../utils/sum'

const fn = vi.fn()

beforeAll(() => {
  vi.stubGlobal(
    'getConfig',
    fn(() => ({
      publicRuntimeConfig: {
        NODE_ENV: 'development',
        ENV: 'development',
      },
    })),
  )

  fn('getConfig', {
    publicRuntimeConfig: {
      NODE_ENV: 'development',
      ENV: 'development',
    },
  })
})

test('home', () => {
  expect(sum(1, 2)).toBe(-1)
})
```

But this didn't work as I expected. So help anyone can help me with this problem.

**Comment by sheremet-va** [comment, marked as answer]
`vi.stubGlobal` does this:
```ts
globalThis[key] = value
```
maybe you need 
```ts
vi.mock("nuxt/config", () => {
  return {
    default: () => ({
      publicRuntimeConfig: {
        NODE_ENV: 'development',
        ENV: 'development',
      },
    })
})
```

**Reply by nhattran998** [reply]
OMG 😆 Thank you very much, really appreciate that.
